shared/src/main/scala/fr/hmil/roshttp/HttpRequest.scala

```diff
@@ -248,31 +248,34 @@ final class HttpRequest  private (
     *
     * @return A future of HttpResponse which may fail with an [[HttpNetworkError]] or [[HttpResponseError]]
     */
-  def send(): Future[HttpResponse] = HttpDriver.send(this)
+  def send(implicit ec: ExecutionContext): Future[HttpResponse] = HttpDriver.send(this)
```

Comment:
To preserve the `.send()` syntax at call side (which should be done IMO given that is sending is a side-effecting operation), this should be defined as
```scala
def send()(implicit ec: ExecutionContext): Future[HttpResponse] = HttpDriver.send(this)
```